### PR TITLE
add short-circuit analysis parameters handling

### PR DESCRIPTION
**Pull request by AbdelHedhili**
Signed-off-by: Abdelsalem <abdelsalem.hedhili@rte-france.com>

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/repository/ShortCircuitParametersEntity.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    public ShortCircuitParametersEntity(boolean withLimitViolations, boolean withVoltageMap, boolean withFeederResult, StudyType studyType, double minVoltageDropProportionalThreshold) {
+        this(null, withLimitViolations, withVoltageMap, withFeederResult, studyType, minVoltageDropProportionalThreshold);
+    }
+
+    @Id
+    @GeneratedValue(strategy  =  GenerationType.AUTO)
+    @Column(name = "id")
+    private UUID id;
+
+    @Column(name = "withLimitViolations", columnDefinition = "boolean default true")
```
Comment (laurgarn) [inline src/main/java/org/gridsuite/study/server/repository/ShortCircuitParametersEntity.java:36]: Why do you repeat the name of the column ?

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 2]
```diff
@@ -481,9 +484,9 @@ public void deleteEquipmentIndexes(UUID networkUuid) {
     }
 
     public CreatedStudyBasicInfos insertStudy(UUID studyUuid, String userId, NetworkInfos networkInfos,
-            String caseFormat, UUID caseUuid, boolean casePrivate, String caseName, LoadFlowParametersEntity loadFlowParameters, UUID importReportUuid) {
+                                              String caseFormat, UUID caseUuid, boolean casePrivate, String caseName, LoadFlowParametersEntity loadFlowParameters, ShortCircuitParametersEntity shortCircuitParametersEntity, UUID importReportUuid) {
```
Comment (laurgarn) [inline src/main/java/org/gridsuite/study/server/service/StudyService.java:487]: You realigned the line. I don't know if it was intented.

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 3]
```diff
@@ -79,7 +79,11 @@ databaseChangeLog:
   - include:
       file: changesets/changelog_2022-08-23T14:02:44Z.xml
       relativeToChangelogFile: true
-
+
```
Comment (laurgarn) [inline src/main/resources/db/changelog/db.changelog-master.yaml:82]: You added some blank spaces on this line

File: src/main/java/org/gridsuite/study/server/repository/ShortCircuitParametersEntity.java (hunk 1 above)
Comment (AbdelHedhili) [reply src/main/java/org/gridsuite/study/server/repository/ShortCircuitParametersEntity.java:36]: no reason in particular, just to be consistent with all the places we used this annotation

File: src/main/java/org/gridsuite/study/server/service/StudyService.java (hunk 2 above)
Comment (AbdelHedhili) [reply src/main/java/org/gridsuite/study/server/service/StudyService.java:487]: fixed

File: src/main/resources/db/changelog/db.changelog-master.yaml (hunk 3 above)
Comment (AbdelHedhili) [reply src/main/resources/db/changelog/db.changelog-master.yaml:82]: fixed

**Review by laurgarn — APPROVED** [review]
Nothing suspect spotted

**Review by etiennehomer — COMMENTED** [review]
Tests OK

File: src/test/java/org/gridsuite/study/server/CaseTest.java [hunk 4]
```diff
@@ -145,7 +147,14 @@ private StudyEntity insertDummyStudy(UUID networkUuid, UUID caseUuid, String cas
                 .dcUseTransformerRatio(true)
                 .hvdcAcEmulation(true)
                 .build();
-        StudyEntity studyEntity = TestUtils.createDummyStudy(networkUuid, caseUuid, caseName, "", defaultLoadflowProvider, defaultLoadflowParametersEntity);
+        ShortCircuitParametersEntity defaultShortCircuitParametersEntity = ShortCircuitParametersEntity.builder()
+                .studyType(StudyType.STEADY_STATE)
```
Comment (etiennehomer) [inline src/test/java/org/gridsuite/study/server/CaseTest.java:151]: toEntity(ShortCircuit.getDefaultParams()) ?

File: src/test/java/org/gridsuite/study/server/LoadflowTest.java [hunk 5]
```diff
@@ -363,7 +365,14 @@ private StudyEntity insertDummyStudy(UUID networkUuid, UUID caseUuid) {
                 .dcUseTransformerRatio(true)
                 .hvdcAcEmulation(true)
                 .build();
-        StudyEntity studyEntity = TestUtils.createDummyStudy(networkUuid, caseUuid, "", defaultLoadflowProvider, defaultLoadflowParametersEntity);
+        ShortCircuitParametersEntity defaultShortCircuitParametersEntity = ShortCircuitParametersEntity.builder()
```
Comment (etiennehomer) [inline src/test/java/org/gridsuite/study/server/LoadflowTest.java:368]: toEntity(ShortCircuit.getDefaultParams()) ?

File: src/test/java/org/gridsuite/study/server/NetworkMapTest.java [hunk 6]
```diff
@@ -383,8 +385,14 @@ private StudyEntity insertDummyStudy(UUID networkUuid, UUID caseUuid) {
             .balanceType(LoadFlowParameters.BalanceType.PROPORTIONAL_TO_GENERATION_P_MAX)
             .connectedComponentMode(LoadFlowParameters.ConnectedComponentMode.MAIN)
             .build();
-
-        StudyEntity studyEntity = TestUtils.createDummyStudy(networkUuid, caseUuid, "", defaultLoadflowProvider, defaultLoadflowParametersEntity);
+        ShortCircuitParametersEntity defaultShortCircuitParametersEntity = ShortCircuitParametersEntity.builder()
```
Comment (etiennehomer) [inline src/test/java/org/gridsuite/study/server/NetworkMapTest.java:388]: toEntity(ShortCircuit.getDefaultParams()) ?

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java [hunk 7]
```diff
@@ -1982,7 +1984,14 @@ private StudyEntity insertDummyStudy(UUID networkUuid, UUID caseUuid, String cas
                 .dcUseTransformerRatio(true)
                 .hvdcAcEmulation(true)
                 .build();
-        StudyEntity studyEntity = TestUtils.createDummyStudy(networkUuid, caseUuid, caseFormat, defaultLoadflowProvider, defaultLoadflowParametersEntity);
+        ShortCircuitParametersEntity defaultShortCircuitParametersEntity = ShortCircuitParametersEntity.builder()
```
Comment (etiennehomer) [inline src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:1987]: toEntity(ShortCircuit.getDefaultParams()) ?

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**